pkgs/by-name/do/dorion/package.nix

```diff
@@ -1,65 +1,186 @@
-{ lib
-, stdenv
-, fetchurl
-, autoPatchelfHook
-, dpkg
-, glib-networking
-, gst_all_1
-, libappindicator
-, libayatana-appindicator
-, webkitgtk_4_0
-, wrapGAppsHook3
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  fetchurl,
+  rustPlatform,
+  autoPatchelfHook,
+  wrapGAppsHook3,
+  glib-networking,
+  gst_all_1,
+  libappindicator,
+  libayatana-appindicator,
+  nodejs,
+  openssl,
+  pkg-config,
+  yq-go,
+  pnpm,
+  webkitgtk_4_1,
+  makeDesktopItem,
+  copyDesktopItems,
 }:
 
-stdenv.mkDerivation (finalAttrs: {
+let
   pname = "dorion";
-  version = "5.0.1";
+  version = "6.2.0";
 
-  src = fetchurl {
-    url = "https://github.com/SpikeHD/Dorion/releases/download/v${finalAttrs.version }/Dorion_${finalAttrs.version}_amd64.deb";
-    hash = "sha256-cCZikTZ+IU3mq/FkJfeggXLyWIsWG+a2qu1GbgW93WQ=";
+  webkitgtk_4_1' = webkitgtk_4_1.override { enableExperimental = true; };
+
+  src = fetchFromGitHub {
+    owner = "SpikeHD";
+    repo = "Dorion";
+    rev = "v${version}";
+    hash = "sha256-7isOL/853IMOUkepbu81INNFgjTHpo9dyvPaxMkBU1U=";
+
+    nativeBuildInputs = [ pnpm.configHook ];
   };
 
-  unpackCmd = ''
-    dpkg -X $curSrc .
-  '';
+  shelter = fetchurl {
+    url = "https://raw.githubusercontent.com/uwu/shelter-builds/c6fb071f74089639a7b2b6177144d31192cbf916/shelter.js";
+    hash = "sha256-2zMvRITkgNnLJvKYtdMHwyXKaC8yIwjUezqYmsihz+o=";
+    meta = {
+      homepage = "https://github.com/uwu/shelter";
+      sourceProvenance = [ lib.sourceTypes.binaryBytecode ]; # actually, minified JS
+      license = lib.licenses.cc0;
+    };
+  };
 
-  runtimeDependencies = [
-    glib-networking
-    libappindicator
-    libayatana-appindicator
-  ];
+  frontend = stdenv.mkDerivation {
```

Comment:
I wouldn't recommend splitting this derivation. In most cases, it's much easier to pull pnpmDeps, pnpm.configHook, etc., into the main derivation and let cargo-tauri build as intended. You can see an example of this in the [test app](https://github.com/NixOS/nixpkgs/blob/cd43d6867192ddfb98ba106065a89790e2722b3f/pkgs/by-name/ca/cargo-tauri/test-app.nix)